pc-bufsw.el

```diff
@@ -338,10 +346,19 @@ Buffers are odered from most to least recently used.")
     str))
 
 (defun pc-bufsw--show-name (i at-left-edge)
-  (let* ((name (buffer-name (aref pc-bufsw--walk-vector i)))
+  (let* ((buf (aref pc-bufsw--walk-vector i))
+	 (name (buffer-name buf))
 	 (current (= i pc-bufsw--cur-index))
-	 (face (if current pc-bufsw-selected-buffer-face
-		 pc-bufsw-buffer-face)))
+	 (face (cond
+		((and pc-bufsw-selected-buffer-face current)
+		 pc-bufsw-selected-buffer-face)
+		((and pc-bufsw-window-buffer-face
+		      (memq buf (mapcar #'car (window-prev-buffers))))
+		 pc-bufsw-window-buffer-face)
+		((and pc-bufsw-frame-buffer-face
+		      (memq buf (frame-parameter (selected-frame) 'buffer-list)))
```

Comment:
In principle this is O(N^2) algorithm again, but that only hits if the users starts to press Ctrl-Tab 100 times. So this OK, but require a comment.

Also, do frame/window lists change during pc-bufsw navigation? And if they change, it will be rather annoying. So I think it is better to cache the membership in another vector that should be stable during navigation.

Reply:
pc-bufsw should not allow them to change. Previewing a buffer should ideally have no side effects.

Reply:
OK, then just add comments about the stability of the lists.